esp-wifi/src/compat/common.rs

```diff
@@ -60,33 +113,30 @@ unsafe extern "C" fn strnlen(chars: *const u8, maxlen: usize) -> usize {
 
 pub fn sem_create(max: u32, init: u32) -> *mut c_void {
     critical_section::with(|_| unsafe {
-        let mut res = 0xffff;
-        memory_fence();
-        for (i, sem) in CURR_SEM.iter().enumerate() {
-            memory_fence();
-            if sem.is_none() {
-                res = i;
-                break;
-            }
-        }
-
-        trace!("sem created res = {} (+1)", res);
+        let ptr = malloc(4) as *mut u32;
+        ptr.write_volatile(init);
 
-        if res != 0xffff {
-            memory_fence();
-            CURR_SEM[res] = Some(init);
-            (res + 1) as *mut c_void
-        } else {
-            core::ptr::null_mut()
-        }
+        trace!("sem created res = {:?}", ptr);
+        ptr.cast()
     })
 }
 
 pub fn sem_delete(semphr: *mut c_void) {
     trace!(">>> sem delete");
+
+    // TODO remove this once fixed in esp_supplicant AND we updated to the fixed
```

Comment:
Might be helpful to add the JIRA ticket number here so we can confirm its resolved later.